controller/internal/controller/consumer_controller.go

```diff
@@ -181,3 +199,76 @@ func (r *ConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	return controller.Complete(r)
 }
+
+// checkConsumerConflicts Perform full conflict detection and rebuild the index
+func (r *ConsumerReconciler) checkConsumerConflicts(ctx context.Context, state *consumerReconcileState) {
+	if r.KeyIndex == nil {
```

Comment:
We need to always create KeyIndex at the beginning. It's not concurrency safe to create it when reconciling.

Reply:
We can provide a similar method like this one: https://github.com/mosn/htnn/blob/4568dd3888f17d275b2a73c6ee66683ec19796f7/controller/internal/controller/filterpolicy_controller.go#L67C6-L67C31